dalle2_pytorch/train_configs.py

```diff
@@ -16,7 +16,43 @@ def default(val, d):
 def ListOrTuple(inner_type):
     return Union[List[inner_type], Tuple[inner_type]]
 
-# pydantic classes
+# general pydantic classes
+
+class TrainSplitConfig(BaseModel):
+    train: float = 0.75
+    val: float = 0.15
+    test: float = 0.1
+
+    @root_validator
+    def validate_all(cls, fields):
+        actual_sum = sum([*fields.values()])
+        if actual_sum != 1.:
+            raise ValueError(f'{fields.keys()} must sum to 1.0. Found: {actual_sum}')
+        return fields
+
+class TrackerConfig(BaseModel):
+    tracker_type: str = 'console'           # Decoder currently supports console and wandb
+    data_path: str = './models'             # The path where files will be saved locally
+    init_config: Dict[str, Any] = None
+    wandb_entity: str = ''                  # Only needs to be set if tracker_type is wandb
+    wandb_project: str = ''
+    verbose: bool = False                   # Whether to print console logging for non-console trackers
+
+# diffusion prior pydantic classes
+
+class AdapterConfig(BaseModel):
+    make: str = "openai"
+    model: str = "ViT-L/14"
+
+    def create(self):
+        if self.make == "openai":
+            return OpenAIClipAdapter(self.model)
+        elif self.make == "x-clip":
+            return XClipAdapter(self.model)
```

Comment:
the factory may need to take in some `base_model_kwargs` and splat it into `XClip`

Reply:
Should `base_model_kwargs` be very explicit? and only for XClip?
```python
class AdapterConfig(BaseModel):
    make: str = "openai"
    model: str = "ViT-L/14"
    base_model_kwargs: Dict[str, Any] = None

    def create(self):
        if self.make == "openai":
            return OpenAIClipAdapter(self.model)
        elif self.make == "x-clip":
            return XClipAdapter(self.model, **self.base_model_kwargs)
        elif self.make == "cocoa":
            return CoCaAdapter(self.model)
        else:
            raise AttributeError("No adapter with that name is available.")
```

Or (perhaps a little less pydantic), allow for extra configs and trust the user will pass meaningful values...
```python
class AdapterConfig(BaseModel):
    make: str = "openai"
    model: str = "ViT-L/14"

    class Config:
        extra = "allow"

    def create(self):
        kwargs = self.dict()
        make = kwargs.pop("make")
        model = kwargs.pop("model")

        if make == "openai":
            return OpenAIClipAdapter(model, **kwargs)
        elif make == "x-clip":
            return XClipAdapter(model, **kwargs)
        elif make == "cocoa":
            return CoCaAdapter(model, **kwargs)
        else:
            raise AttributeError("No adapter with that name is available.")
```

Reply:
yea, it is tricky because `XClipAdapter` actually wraps a `XClip` model, (and similarly `CoCaAdapter` wraps `Coca`) perhaps it is best to just leave them unsupported for now

Reply:
I think I see what you were getting at now, I believe I fixed it! 

If that still isn't what you had in mind, lmk, I can revert and we can just leave it unsupported for now